client/containers/Pub/PubHeader/SmallHeaderButton.tsx

```diff
@@ -1,39 +1,36 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { Button } from 'reakit';
 
 import { Icon } from 'components';
 
 require('./smallHeaderButton.scss');
 
-const propTypes = {
-	className: PropTypes.string,
-	disabled: PropTypes.bool,
-	href: PropTypes.string,
-	icon: PropTypes.string.isRequired,
-	label: PropTypes.string,
-	labelPosition: PropTypes.oneOf(['left', 'right']),
-	onClick: PropTypes.func,
-	tagName: PropTypes.string,
+type Props = {
+	className?: string;
+	disabled?: boolean;
+	href?: null | string;
+	icon: React.ReactNode;
+	label?: null | string;
+	labelPosition?: 'left' | 'right';
+	onClick?: () => unknown;
+	tagName?: string;
 };
 
-const defaultProps = {
-	className: '',
-	disabled: false,
-	href: null,
-	label: null,
-	onClick: null,
-	tagName: 'button',
-	labelPosition: 'left',
-};
-
-const SmallHeaderButton = React.forwardRef((props, ref) => {
-	// @ts-expect-error ts-migrate(2339) FIXME: Property 'className' does not exist on type '{ chi... Remove this comment to see the full error message
-	const { className, disabled, href, icon, label, labelPosition, onClick, tagName } = props;
+const SmallHeaderButton = React.forwardRef((props: Props, ref) => {
+	const {
+		className = '',
+		disabled = false,
+		href = null,
+		icon,
+		label = null,
+		labelPosition = 'left',
+		onClick = null,
+		tagName = 'button',
+	} = props;
 	return (
 		<Button
-			as={tagName}
+			as={tagName as any}
```

Comment:
Just out of curiosity, what error were you getting here?

Reply:
I don't know where to find a union of all possible HTML tag names in TS (though I assume React would export one — this is like 70% laziness and 30% confusion)